adapter-base/src/test/java/org/eclipse/hono/adapter/auth/device/x509/OCSPIntegrationTest.java

```diff
@@ -0,0 +1,239 @@
+        throws CertificateException, IOException {
+        final RevocableTrustAnchor anchor = createTrustAnchor();
+        final X509Certificate revokedCert = loadCertificate("ocsp/ocsp-revoked.pem");
+
+        validator.validate(List.of(revokedCert), anchor).onComplete(ctx.succeedingThenComplete());
+    }
+
+    /**
+     * Verifies that the validator succeeds to verify a valid certificate when OCSP
+     * revocation check is enabled.
+     *
+     * @param ctx The vert.x test context.
+     */
+    @Test
+    void testValidateSucceedsForValidOCSPCertificate(final VertxTestContext ctx) throws CertificateException, IOException {
+        final RevocableTrustAnchor anchor = createTrustAnchor();
+        final X509Certificate caCert = loadCertificate("ocsp/cacert.pem");
+        anchor.setOcspEnabled(true);
+        anchor.setOcspResponderUri(getOCSPResponderUri());
+        anchor.setOcspResponderCert(caCert);
+        anchor.setOcspNonceEnabled(true);
+        final X509Certificate validCert = loadCertificate("ocsp/ocsp-valid.pem");
+
+        validator.validate(List.of(validCert), anchor).onComplete(ctx.succeedingThenComplete());
+    }
+
+    /**
+     * Verifies that the validator succeeds to verify a valid certificate when OCSP
+     * revocation check is enabled and another trust anchor without OCSP is configured.
+     *
+     * @param vertx The vert.x instance to use.
+     * @param ctx The vert.x test context.
+     */
+    @Test
+    void testValidateSucceedsWithAnotherAnchorWithoutOCSP(final Vertx vertx, final VertxTestContext ctx) throws CertificateException, IOException {
+        final RevocableTrustAnchor anchor = createTrustAnchor();
+        final X509Certificate caCert = loadCertificate("ocsp/cacert.pem");
+        anchor.setOcspEnabled(true);
+        anchor.setOcspResponderUri(getOCSPResponderUri());
+        anchor.setOcspResponderCert(caCert);
+        anchor.setOcspNonceEnabled(true);
+        final X509Certificate validCert = loadCertificate("ocsp/ocsp-valid.pem");
+
+        final SelfSignedCertificate anotherCaCert = SelfSignedCertificate.create("example.com");
+        VertxTools.getCertificate(vertx, anotherCaCert.certificatePath())
+            .compose((cert) -> {
+                final TrustAnchor anotherAnchor = new RevocableTrustAnchor(cert.getSubjectX500Principal(),
+                    cert.getPublicKey(), null);
+                return validator.validate(List.of(validCert), Set.of(anotherAnchor, anchor));
+            })
+            .onComplete(ctx.succeedingThenComplete());
+    }
+
+    /**
+     * Verifies that untrusted certificate fails to verify also in case that OCSP is enabled.
+     *
+     * @param vertx The vert.x instance to use.
+     * @param ctx The vert.x test context.
+     */
+    @Test
+    void testValidateSucceedsForTrustAnchorBasedOnPublicKey(final Vertx vertx, final VertxTestContext ctx) throws CertificateException, IOException {
```

Comment:
are you sure about the naming of this test case? The code looks like you are expecting the validation to fail ...